core-bundle/contao/drivers/DC_Table.php

```diff
@@ -6401,7 +6401,22 @@ protected function filterMenu($intFilterPanel)
 					// Associative array
 					elseif (($GLOBALS['TL_DCA'][$this->strTable]['fields'][$field]['eval']['isAssociative'] ?? null) || ArrayUtil::isAssoc($GLOBALS['TL_DCA'][$this->strTable]['fields'][$field]['options'] ?? null))
 					{
-						$option_label = $GLOBALS['TL_DCA'][$this->strTable]['fields'][$field]['options'][$vv] ?? null;
+						$option_label = null;
+
+						if (isset($GLOBALS['TL_DCA'][$this->strTable]['fields'][$field]['options'][$vv]))
+						{
+							$option_label = $GLOBALS['TL_DCA'][$this->strTable]['fields'][$field]['options'][$vv];
+						}
+						else
+						{
+							// Handle grouped options (see #7409)
+							$current = current($GLOBALS['TL_DCA'][$this->strTable]['fields'][$field]['options']);
```

Comment:
I don't think this would work. Wouldn't we need to go through all groups to find a matching option?

Reply:
Of course 🤦‍♂️